rom/dev/doc/threat-model.md

```diff
@@ -0,0 +1,94 @@
+# Outline
+The documentation’s purpose is to iterate threat that the Caliptra ROM will face from potential attackers, we will also look at the potential mitigations and the difficulty level of the attack. The goal for this document is to look at just the Caliptra ROM and what thread that it can potentially face.  There is a comprehensive [Caliptra Asset & Threats](https://github.com/chipsalliance/Caliptra/blob/main/doc/Caliptra.md#caliptra-assets--threats) writing currently in the Caliptra specification that covers Caliptra IP as a whole. 
+The main area of attack surface that we will look at includes the following,
+-	ROM Confidentiality / Reverse Engineering Attack
+-	Mailbox Interface Attack
+-	Physical Attacks (Glitching)
+-	DPA/DFA Attacks
+-	Oracle Attacks
+
+# ROM Confidentiality / Reverse Engineering Attack
+
+## Attack Description
+Usually, the ROM of a silicon’s ROT is considered confidential for many reasons, especially to prevent reverse engineering attempts that can potentially lead to more exploitable vulnerabilities from white box attacks.  Since Caliptra’s soul is an Open-Source Project and the ROM code is available for all to see.  There is no confidentiality in the ROM code at all and we need to consider all types of white box attacks.  Attackers are armed with full knowledge of the ROM code so any logical error might be able to be turned into an exploitable vulnerability.
+
+## Mitigation
+Mitigation for this will consist mostly typical code review and analysis methods.
+- Rigorous code review 
+  - The best defense is to ensure that the ROM code does not have exploitable vulnerability that can be exported.  Code review and finding coding error early on can help reduce vulnerability vectors that might be able to be exported by a malicious actor.
+- Proactive static analysis 
+  - Static analysis would be a great tool to employe in mitigating this.  Even though Rust does have compile-time error checks, it can still benefit from running rust focused static analysis tools to further reduce potential coding errors.
```

Comment:
We're currently using cargo clippy's defaults lints (all warnings are errors):

https://github.com/rust-lang/rust-clippy

Reply:
Hi Kor, the goal for this is to list otu the thread and potential mitigations.  I think after we agree that this is the list of threat we care about we would indicate how they are addressed.  For example this threat can be somewhat addressed by using the werror equivalent for rust.  There are other static analysis strategy for RUST that might further reduce risk if we plan on using them.  However the goal for getting this PR merged is to just identify and list out the threats we care about